### PR TITLE
Main landing page

### PR DESCRIPTION
**Pull request by rmzse**
## Pivotal Tracker story 

https://www.pivotaltracker.com/story/show/137256271

## Changes proposed in this pull request:

This feature creates the main landing page

## What I have learned working on this feature:

• Recap of the basic cycle of routes, controllers and views. 
• Separating assertion steps from other steps
• Check the existence of specific DOM elements of certain types with certain names and id's
• Rails tags for input types and divs

#### Ready for review!

### CODE REVIEW

**Comment by rmzse** [comment]
@Philippoes @aniansson pls review and approve in order to continue your work!

**Review by aniansson — APPROVED** [review]

**Review by Philippoes — APPROVED** [review]
Looks good!

**Comment by rmzse** [comment]
@Philippoes @aniansson now this is ready to merge!

File: app/views/landing/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<%= content_tag(:div, "Logotype", id: ["logotype"]) %>
```
Comment (tochman) [inline app/views/landing/index.html.erb]: Why is the `id:` wrapped in hard brackets?

File: features/step_definitions/assertion_steps.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+end
+
+Then(/^I should see a "([^"]*)" placeholder$/) do |id|
+  expect(page).to have_selector(:css, "div##{id}")
+end
+
+Then(/^I should see a link called "([^"]*)" that links to the main landing page$/) do |link|
+  expect(page).to have_link link
+  click_link link
+	%q{ When I visit the main landing page }
```
Comment (tochman) [inline features/step_definitions/assertion_steps.rb]: No, no, no... we don't mix assertions with interaction commands

**Review by tochman — CHANGES_REQUESTED** [review]
See comments....

File: app/views/landing/index.html.erb (hunk 1 above)
Comment (rmzse) [reply app/views/landing/index.html.erb]: According to the finders dox that's the way to do it, but removing them works fine too. =)

File: features/main_landing_page.feature [hunk 3]
```diff
@@ -0,0 +1,12 @@
+Feature: Main landing page
+  As a visitor
+  In order to get an overview over the service
+  I need to see a main landing page
+
+  Scenario: Visitor visits main page
+    When I visit the main landing page
+    Then I should see "Welcome to HomeChef"
+    And I should see input of type "text" with name "enter_location"
+    And I should see a placeholder for dishes in general with an id of "random_dishes"
```
Comment (tochman) [inline features/main_landing_page.feature]: I would use `I should see a "random_dishes" placeholder`. This feels like a duplicate.

**Comment by rmzse** [comment]
@tochman code review pls!

File: features/main_landing_page.feature [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  In order to get an overview over the service
+  I need to see a main landing page
+
+  Scenario: Visitor visits main page
+    When I visit the main landing page
+    Then I should see "Welcome to HomeChef"
+    And I should see input of type "text" with name "enter_location"
+    And I should see a "general_random_dishes" placeholder
+    And I should see a "logotype" placeholder
+    And I should see a link called "Home"
```
Comment (AmberWilkie) [inline features/main_landing_page.feature]: This step is totally unnecessary, considering you're clicking the link in the next step.

File: features/step_definitions/assertion_steps.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  expect(page).to have_selector(:css, "div##{id}")
+end
+
+Then(/^I should see a link called "([^"]*)"$/) do |name|
+  expect(page).to have_link name
+end
+
+Then(/^I should go to the main landing page$/) do
+  %q{ When I visit the main landing page }
+end
```
Comment (AmberWilkie) [inline features/step_definitions/assertion_steps.rb]: Dudes. It doesn't have to be this way. If you insist on having different wording, you can do instead:
`I should (?: go to | be on) the main landing page` and then the content of the step is the same.

File: features/step_definitions/main_landing_page_steps.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+When(/^I visit the main landing page$/) do
+  visit root_path
+end
+
+When(/^I click the "([^"]*)" link$/) do |link|
+  click_link link
+  %q{ I visit the main landing page }
```
Comment (AmberWilkie) [inline features/step_definitions/main_landing_page_steps.rb]: No no no. If you click the link and it takes you to the main landing page, then you should be on the main landing page. You're clicking a link and then redirecting in the same step.

File: features/main_landing_page.feature (hunk 4 above)
Comment (rmzse) [reply features/main_landing_page.feature]: Yes, we had the link and the click of it in one statement before, but were required to split them. Should we omit the naming of the link we are clicking altogether? How can we make sure it is THAT specific one link that leads us to that specific page?

File: features/step_definitions/assertion_steps.rb (hunk 5 above)
Comment (rmzse) [reply features/step_definitions/assertion_steps.rb]: Great stuff! Let's change that IMMEDIATELY.

**Comment by AmberWilkie** [comment]
You are already testing that the link exists, simply by clicking it. If
there's no link, Cucumber will say "link not found" or whatever. You don't
have to directly test that there is a link (vs. text).


*Amber Wilkie*
www.amberwilkie.com
0720.08.27.07 | Instagram <http://www.instagram.com/amberwphoto>

File: features/main_landing_page.feature (hunk 4 above)
Comment (AmberWilkie) [reply features/main_landing_page.feature]: You do not have to test that the link is on the page if you are clicking the link. If there was no link, you couldn't click it.

File: features/step_definitions/assertion_steps.rb (hunk 5 above)
Comment (rmzse) [reply features/step_definitions/assertion_steps.rb]: That solution did not work for us, we simplify it and keep two statements for visiting root path.

**Comment by rmzse** [comment]
@AmberWilkie code review pls!

**Review by tochman — APPROVED** [review]

**Review by AmberWilkie — APPROVED** [review]
Much better!

**Pull request merged**